src/osd/PrimaryLogPG.cc

```diff
@@ -15751,7 +15751,11 @@ int PrimaryLogPG::getattr_maybe_cache(
 	*val = i->second;
       return 0;
     } else {
-      return -ENODATA;
+      if (obc->obs.exists) {
+        return -ENODATA;
+      } else {
+        return -ENOENT;
```

Comment:
PrimaryLogPG.cc:6420 shows
if (result < 0 && result != -EEXIST && result != -ENODATA)

Is there a confusion here between EEXIST & ENOENT?

Reply:
@ronen-fr the ENODATA and ENOENT will returned in case we have attribute in cache, otherwise will check the store (that probably can return EEXIST.)

Reply:
@NitzanMordhai : to clarify my question:
Seems like there are 3 possible error values that might be returned, right?
EEXIST, ENOENT & ENODATA
but the caller specifically handles only two of those. My question is - was that intentional?